osquery/tables/applications/posix/docker.cpp

```diff
@@ -0,0 +1,866 @@
+  r["memory_limit"] =
+      (tree.get<bool>("MemoryLimit", false) ? INTEGER(1) : INTEGER(0));
+  r["swap_limit"] =
+      (tree.get<bool>("SwapLimit", false) ? INTEGER(1) : INTEGER(0));
+  r["kernel_memory"] =
+      (tree.get<bool>("KernelMemory", false) ? INTEGER(1) : INTEGER(0));
+  r["cpu_cfs_period"] =
+      (tree.get<bool>("CpuCfsPeriod", false) ? INTEGER(1) : INTEGER(0));
+  r["cpu_cfs_quota"] =
+      (tree.get<bool>("CpuCfsQuota", false) ? INTEGER(1) : INTEGER(0));
+  r["cpu_shares"] =
+      (tree.get<bool>("CPUShares", false) ? INTEGER(1) : INTEGER(0));
+  r["cpu_set"] = (tree.get<bool>("CPUSet", false) ? INTEGER(1) : INTEGER(0));
+  r["ipv4_forwarding"] =
+      (tree.get<bool>("IPv4Forwarding", false) ? INTEGER(1) : INTEGER(0));
+  r["bridge_nf_iptables"] =
+      (tree.get<bool>("BridgeNfIptables", false) ? INTEGER(1) : INTEGER(0));
+  r["bridge_nf_ip6tables"] =
+      (tree.get<bool>("BridgeNfIp6tables", false) ? INTEGER(1) : INTEGER(0));
+  r["oom_kill_disable"] =
+      (tree.get<bool>("OomKillDisable", false) ? INTEGER(1) : INTEGER(0));
+  r["logging_driver"] = tree.get<std::string>("LoggingDriver", "");
+  r["cgroup_driver"] = tree.get<std::string>("CgroupDriver", "");
+  r["kernel_version"] = tree.get<std::string>("KernelVersion", "");
+  r["os"] = tree.get<std::string>("OperatingSystem", "");
+  r["os_type"] = tree.get<std::string>("OSType", "");
+  r["architecture"] = tree.get<std::string>("Architecture", "");
+  r["cpus"] = INTEGER(tree.get<int>("NCPU", 0));
+  r["memory"] = BIGINT(tree.get<uint64_t>("MemTotal", 0));
+  r["http_proxy"] = tree.get<std::string>("HttpProxy", "");
+  r["https_proxy"] = tree.get<std::string>("HttpsProxy", "");
+  r["no_proxy"] = tree.get<std::string>("NoProxy", "");
+  r["name"] = tree.get<std::string>("Name", "");
+  r["server_version"] = tree.get<std::string>("ServerVersion", "");
+  r["root_dir"] = tree.get<std::string>("DockerRootDir", "");
+  results.push_back(r);
+
+  return results;
+}
+
+/**
+ * @brief Utility method to create query arguments for docker API URI.
+ *
+ * @param context Query context which contains SQL constraint.
+ * @param key Constraint column to look for (eg: "id", "name").
+ * @param query Placeholder for returning query string.
+ * @param set Set for returning constraint values for specified key.
+ * @param add_all Whether to add "all=1" to query string or not.
+ */
+void getQuery(QueryContext& context,
+              const std::string& key,
+              std::string& query,
+              std::set<std::string>& set,
+              bool add_all) {
+  if (context.constraints[key].exists(EQUALS)) {
+    std::string key_str;
+    std::set<std::string> const_set = context.constraints[key].getAll(EQUALS);
+    set.insert(const_set.begin(), const_set.end());
+
+    for (const auto& item : set) {
+      if (!key_str.empty()) {
+        key_str.append("%2C"); // comma
+      }
+      key_str.append("%22").append(item).append("%22%3Atrue"); // "item":true
```

Comment:
This seems scary, can I inject arbitrary content into the API call with a where clause?